### ISSUE TITLE
Irony does not certain file types

### ISSUE THREAD

**Issue by 3246251196** [Sarcasm/irony-mode#248]
Potentially linked to 163 and 206 

Irony does not parse on these file types (this is running unix file):
    ISO-8859 C program text, with CRLF line terminators
    Non-ISO extended-ASCII English text, with CRLF line terminators

When attempting this and then trying to run my manual complete (company-irony) I get thing like:

    Error in post-command-hook (irony-completion--post-command): (file-error "Writing to process" "bad file descriptor" #<process Irony>)

Then, killing irony and requesting another manual complete(company-irony)
    Company: An error occurred in auto-begin
    Company: Back-end company-irony async timeout with args (candidates )

Either way, this all stems from the problems with these two file types.

This is on Windows 64bit, 
    ./irony-server.exe --version
    irony-server version 0.2.0-cvs4

**Comment by Sarcasm** [comment]
Hum, I was suspecting that it was some encoding issues with the process communication but it does not appear to work when using irony-server directly from the command line, so I guess the issue is purely in the C++. I do not have much ideas about what is causing the issue. Maybe one should start from a dead-simple example of using libclang, hopefully the behavior will not appear, then try to understand the difference with irony-mode. I suspect I may be doing something wrong with the libclang API but no idea why.

**Comment by 3246251196** [comment]
I will see if converting the files back to an ASCII type works tomorrow.

**Comment by 3246251196** [comment]
Encoding things back to ASCII does work. A lot of the time it is a matter of ensuring that there is no £ sign in the code - basically anything that is non-ascii.

**Comment by 3246251196** [comment]
Update on this:

I have a simple cpp file. I am alternative between it containing a $ character somewhere (even in just a comment) or a £.

<code>
#include <string>
struct Here
{
   int x;
   int y;
}
/* £ */
int main()
{
   std::string str = "£";
   Here h;
   h.
   return 0;
}
</code>

When I run:

~/.emacs.d/irony/bin/irony-server.exe complete main2.cpp 12 6

I get completions back in the console window. (Of course, irony-server waits for cin to be provided with the compiler flags which I am just giving "-I." to.)

Then, I can replace the £ with a $ and exactly the same thing happens.

I have noticed that the problem only occurs in emacs. So, if I do

emacs main2.cpp

Go to the "h." and then press f11 (which is my version of: company-complete) with the version of the file with $ in it, I get completions. However, if I repeat this when it contains a £ sign I get an infinite loop. So, doesn't this mean that it is something with emacs? irony-server provided completions and exits with 0 status regardless of non-ascii or ascii characters.

Cheers

**Comment by Sarcasm** [comment]
Yeah, I also I don't know if it's an emacs thing or a stdin thing.
I always wondered if it was a multibyte thing, or buffer encoding thing. I would be curious if one of the following things change the behavior:

* `(set-process-coding-system proc 'binary 'binary)`
*  `(set-buffer-multibyte nil)`
* when writing, use `(coding-system-for-write 'raw-text-unix) (write-file ...)`

**Comment by 3246251196** [comment]
I can check this tomorrow and work. 

I should have been clearer above. Without using emacs and just using irony-server from the command line, completions work regardless of encoding. It is just in emacs where the behaviour is occurring. So I will try what you suggested tomorrow.
